### PR TITLE
feat: allow checking an offer is compliant with the fair exchange policy

### PR DESCRIPTION
**Pull request by levalleux-ludo**
## Description
a new service is now available in the core-sdk:
`public async checkExchangePolicy(
    offerId: BigNumberish,
    rules: offers.CheckExchangePolicyRules
): Promise<offers.CheckExchangePolicyResult>`

Example of the rules are available here (env dependent):
[exchangePolicyRules.testing.json](packages\core-sdk\tests\exchangePolicy\exchangePolicyRules.testing.json)
[exchangePolicyRules.staging.json](packages\core-sdk\tests\exchangePolicy\exchangePolicyRules.staging.json)
[exchangePolicyRules.production.json](packages\core-sdk\tests\exchangePolicy\exchangePolicyRules.production.json)

## How to test

- unit tests have been added in core-sdk package
- a script has been added to call the service manually

### CODE REVIEW

File: data/exchangePolicies/fairExchangePolicy-testing.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://example.com/person.schema.json",
+  "title": "Person",
+  "description": "A person",
+  "type": "object",
+  "properties": {}
```
Comment (albertfolch-redeemeum) [inline data/exchangePolicies/fairExchangePolicy-testing.json:7]:
```suggestion
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "http://example.com/person.schema.json",
  "title": "Person",
  "description": "A person",
  "type": "object",
  "properties": {}
```

File: packages/core-sdk/tests/exchangePolicy/exchangePolicyRules.production.json [hunk 2]
```diff
@@ -0,0 +1,107 @@
+{
+  "yupSchema": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "http://example.com/person.schema.json",
+    "title": "Person",
+    "description": "A person",
```
Comment (albertfolch-redeemeum) [inline packages/core-sdk/tests/exchangePolicy/exchangePolicyRules.production.json:6]: I guess these values are not used? otherwise change them I guess

**Review by albertfolch-redeemeum — APPROVED** [review]
LGTM

File: packages/core-sdk/tests/exchangePolicy/exchangePolicyRules.production.json (hunk 2 above)
Comment (levalleux-ludo) [reply packages/core-sdk/tests/exchangePolicy/exchangePolicyRules.production.json]: done

**Pull request merged**